web-client/public/js/setup-handlers.js

```diff
@@ -69,4 +69,13 @@ export const setupHandlers = grnState => {
         updateApp(grnState);
     });
 
+    $("#colorEdges").click(() => {
+        grnState.colorOptimal = true;
+        updateApp(grnState);
+    });
+
+    $("#blackEdges").click(() => {
+        grnState.colorOptimal = false;
+        updateApp(grnState);
+    });
```

Comment:
So yes, these are the handlers which update `grnState` based on user actions on `#colorEdges` and `#blackEdges`, but shouldn't M->V code also exist in `updateApp`? i.e., Set `#colorEdges` and `#blackEdges` to the appropriate state based on the value of `grnState.colorOptimal`?